core/src/main/java/org/eobjects/datacleaner/extension/output/CreateCsvFileAnalyzer.java

```diff
@@ -103,20 +135,135 @@ public String getSuggestedLabel() {
 
     @Override
     public OutputWriter createOutputWriter() {
-        final String[] headers = new String[columns.length];
-        for (int i = 0; i < headers.length; i++) {
-            headers[i] = columns[i].getName();
+        List<String> headers = new ArrayList<String>() ;
+        for (int i = 0; i < columns.length; i++) {
+            String columnName = columns[i].getName();
+            headers.add(columnName);
+            if(columnToBeSortedOn != null) {
+                if(columnName.equals(columnToBeSortedOn.getName())){
+                    indexOfColumnToBeSortedOn = i ;
+                }
+            }
+        }
+
+        if(columnToBeSortedOn != null) {
+            if(indexOfColumnToBeSortedOn == -1) {
+                this.isColumnToBeSortedOnPresentInInput  = false ;
+                indexOfColumnToBeSortedOn = columns.length ;
+                headers.add(columnToBeSortedOn.getName()) ;
+                InputColumn<?>[] newColumns = new InputColumn<?>[columns.length + 1] ;
+                for(int i = 0; i < columns.length; i++){
+                    newColumns[i] = columns[i] ;
+                }
+                newColumns[columns.length]  = columnToBeSortedOn ;
+                columns = newColumns ;
```

Comment:
Very minor: The "columns" field should ideally not change. It is set by the job configuration and should be considered immutable.